src/daft-image/src/ops.rs

```diff
@@ -98,6 +105,96 @@ impl ImageOps for ImageArray {
                 .rename(self.name())),
         }
     }
+
+    fn average_hash(&self) -> DaftResult<Utf8Array> {
+        let mut results = Vec::with_capacity(self.len());
+
+        for i in 0..self.len() {
+            if let Some(img) = self.as_image_obj(i) {
+                let hash = compute_average_hash(img)?;
+                results.push(Some(hash));
+            } else {
+                results.push(None);
+            }
+        }
```

Comment:
no action needed: So i think this is fine for now, but iterating like this is quite expensive.

these are arrow arrays so iterating over it this way also makes you check the validity (Some/None). Instead you could extract the validity & values separately, map the values then reconstruct the array with the validity. No need to unnecessarily check for the Nones. Also this `arrow2::array::Utf8Array::from(results)` can be pretty expensive too. It'd be much more performant avoiding the native rust types all together.